### PR TITLE
fix: don't accept hex colors on clan name, tag and rank

### PR DESCRIPTION
**Pull request by Tomut0**
Thank to https://github.com/RoinujNosde/SimpleClans/pull/363#issuecomment-1529056857 for pointing out a problem.

### CODE REVIEW

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/conversation/CreateClanNamePrompt.java [hunk 1]
```diff
@@ -85,18 +85,20 @@ private void processClanCreation(@NotNull SimpleClans plugin, @NotNull Player pl
     private Prompt validateName(@NotNull SimpleClans plugin, @NotNull Player player, @NotNull String input) {
         boolean bypass = plugin.getPermissionsManager().has(player, "simpleclans.mod.bypass");
         if (!bypass) {
-            if (ChatUtils.stripColors(input).length() > plugin.getSettingsManager().getInt(CLAN_MAX_LENGTH)) {
+            String strippedInput = ChatUtils.stripColors(input);
+            if (strippedInput.length() > plugin.getSettingsManager().getInt(CLAN_MAX_LENGTH)) {
```
[the remark this answers is not in the document]
Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/conversation/CreateClanNamePrompt.java]: This commit is not about a refactor, freaking bot 👺

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/utils/ChatUtils.java [hunk 2]
```diff
@@ -56,6 +56,16 @@ public static String parseColors(@NotNull String text) {
...
     }
 
+    /**
+     * Checks whatever string has a bukkit's {@link org.bukkit.Color} or hex color
+     *
+     * @param text the string
+     * @return true if string has colors, otherwise false.
+     */
+    public static boolean hasColors(@NotNull String text) {
+        return text.contains("&") || HEX_COLOR_SUPPORT && HEX_COLOR_PATTERN.matcher(text).find();
```
Comment (RoinujNosde) [inline src/main/java/net/sacredlabyrinth/phaed/simpleclans/utils/ChatUtils.java]: Isn't the first check enough for hex colors?

Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/utils/ChatUtils.java]: изображение
I'm wonder as well

Comment (RoinujNosde) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/utils/ChatUtils.java]: https://github.com/RoinujNosde/SimpleClans/blob/master/src/main/java/net/sacredlabyrinth/phaed/simpleclans/commands/clan/LeaderCommands.java#L191

Rename doesn't check for colors.

Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/utils/ChatUtils.java]: I see, will fix it later. 
Thanks. :pepe_naruto:

**Pull request closed without merging**